sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/BlockHeight.kt

```diff
@@ -0,0 +1,56 @@
+package cash.z.ecc.android.sdk.model
+
+import cash.z.ecc.android.sdk.type.ZcashNetwork
+
+/**
+ * Represents a block height, which is a UInt32.  SDK clients use this class to represent the "birthday" of a wallet.
+ *
+ * @param value The block height.  Must be in range of a UInt32.
+ */
+/*
+ * For compatibility with Java clients, this class represents the height value as a Long with
+ * assertions to ensure that it is a 32-bit unsigned integer.
+ */
+data class BlockHeight internal constructor(val value: Long) : Comparable<BlockHeight> {
```

Comment:
I'd consider adding decrement helper function given the number of places `height + -1` is used. I haven't noticed occurrences of other decrements that are not just subtracting one.